tests/unit/test_options.py

```diff
@@ -286,6 +287,107 @@ def test_subcommand_option_before_subcommand_fails(self):
             main(['--find-links', 'F1', 'fake'])
 
 
+@contextmanager
+def tmpconfig(option, value, section='global'):
+    with NamedTemporaryFile(mode='w', delete=False) as f:
+        f.write('[{}]\n{}={}\n'.format(section, option, value))
+        name = f.name
+    try:
+        yield name
+    finally:
+        os.unlink(name)
```

Comment:
Why not
```suggestion
def tmpconfig(option, value, section='global'):
    with NamedTemporaryFile(mode='w', delete=True) as f:
        f.write('[{}]\n{}={}\n'.format(section, option, value))
        yield f.name
```

Reply:
I spent a solid minute trying to figure out why I wrote it as I did it :smile:  Turns out it was to ensure that the file is written to disk (which is warrantied by `f.__exit__`).  I think we can use `f.flush()` as well although I'm not sure enough about NamedTemporaryFile semantic to tell if that is safe on all platforms.

Reply:
The file is definitely written on every newline AFAICT, so I think @xavfernandez's suggestion will work well for simplifying this logic. :)

Reply:
FYI writing newline doesn't write to disk (at least not by default), but thanks for raising this up.  The best source I can refer to is [ISO C99 7.19.3.3](http://www.open-std.org/jtc1/sc22/WG14/www/docs/n1256.pdf#%5B%7B%22num%22%3A555%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C-27%2C816%2Cnull%5D) which states:

> When a stream is unbuffered, characters are intended to appear from the source or at the destination as soon as possible. Otherwise characters may be accumulated and transmitted to or from the host environment as a block.
>
> When a stream is fully buffered, characters are intended to be transmitted to or from the host environment as a block when a buffer is filled.
>
> When a stream is line buffered, characters are intended to be transmitted to or from the host environment as a block when a new-line character is encountered.

I dig down CPython docs again and found (NamedTemporaryFile passes `buffering` to [open](https://docs.python.org/3/library/functions.html#open):

> `buffering` is an optional integer used to set the buffering policy. Pass 0 to switch buffering off (only allowed in binary mode), 1 to select line buffering (only usable in text mode), and an integer > 1 to indicate the size in bytes of a fixed-size chunk buffer. When no buffering argument is given, the default buffering policy works as follows:
>
> * Binary files are buffered in fixed-size chunks; the size of the buffer is chosen using a heuristic trying to determine the underlying device’s “block size” and falling back on `io.DEFAULT_BUFFER_SIZE`. On many systems, the buffer will typically be 4096 or 8192 bytes long.
> * “Interactive” text files (files for which `isatty()` returns True) use line buffering. Other text files use the policy described above for binary files.

So if we set `buffering=1` the file would be `flush`ed upon newline, otherwise after a few KiB.  However, `TextIOWrapper.flush` [does not seem warranty to flush the filesystem file](https://github.com/python/cpython/blob/3d17c045b4c3d09b72bbd95ed78af1ae6f0d98d2/Modules/_io/textio.c#L1540-L1541) (or maybe I interpret the comment incorrectly, please don't ask me what the code actually does).

That being said, I'm pretty sure on most platforms it is safe to assume `flush` would do the job and would not object *someone else* applying the patch above.  Personally I just do know enough to be responsible for such change.